### PR TITLE
#75/pagination

### LINKED ISSUE #75: Api pagination 🧙‍♂️
### Expected Behavior

When we call our api for the `ExplorePage` or the `FavouritesPage` we should have pagination in the fetch response to prevent us from trying to load an excessive amount of data in one go

### Current Behavior

It's the wild west baby - just load up as much data as there is 

### Any other notes

_No response_

### Assignment

This issue is free for anyone to take

### PR DESCRIPTION
**Pull request by camelPhonso**
# Description

**Closes #75**  

Sets pagination for the Explore and Favourites page.

This pr refactors some of our query strings and message handlers to receive a `pageNumber` argument to fetch a specific page for the GET call made. This is connected to the frontend with a new component that's been imported to both page components.

Depending on what we decided on the next design review we can change the look of that component in a single place.

### Files changed

- `components/PaginationNav/**` - new component to control the fetching of a new page in the fetch call
- `ExplorePage.tsx` && `FavouritesPage.tsx` - imported the new component and refactored the logic around the fetch of data in the useEffect to handle pagination
- `components/PopupCard/` && `components/ProjectCard` - both components needed a border to match the figma design, so that was added to the image element tag in both
-   `types/baseTypes.ts` - declared a `Meta` interface to allow us to handle the metadata returned from fetch calls that include pagination
- `allProjectsQuery.ts` - now returns a function that creates the query string to include pagination and request the page number passed to it as an argument
- `messages/fetchListOfPopups.ts` && `popupsListQuery.ts` - new message handler that fetches a list of popups based on an array of popup ids && query string dynamically created to return the requested page of the api call

### UI changes

Screenshot 2024-10-29 at 09 31 29
Screenshot 2024-10-29 at 09 32 05

### Changes to Documentation

n/a

# Tests

no new tests -  the existing tests check the return of the same message handlers, we are only isolating a number of items to return at each time

### CODE REVIEW

File: src/components/PaginationNav/PaginationNav.css [hunk 1]
```diff
@@ -0,0 +1,8 @@
+.pagination-nav {
+  display: flex;
+  flex-direction: row;
```
Comment (nichgalzin) [inline src/components/PaginationNav/PaginationNav.css]: small thing, don't need this as it's default value

File: src/components/page-components/ExplorePage/ExplorePage.tsx [hunk 2]
```diff
@@ -9,25 +9,44 @@ import { sendToBackground } from "@plasmohq/messaging"
...
+        
+      setPageCount(meta.pagination.pageCount)
       setProjects(data)
     }
     fetchAllProjects()
-  }, [])
+  }, [pageNumber])
+
+  const navigateToNext = () => {
+    setPageNumber(page => page + 1)
```
Comment (nichgalzin) [inline src/components/page-components/ExplorePage/ExplorePage.tsx]: Another small thing. If you move the click handlers to the PaginationNav component and just pass the setPageNumber function you would not have to declare the functions in both the explore and favourites page prevention a little bit of code duplication.

**Review by nichgalzin — CHANGES_REQUESTED** [review]

File: src/components/page-components/ExplorePage/ExplorePage.tsx (hunk 2 above)
Comment (camelPhonso) [reply src/components/page-components/ExplorePage/ExplorePage.tsx]: uhh yeah, that looks way better

File: src/background/messages/viewSinglePopup.ts [hunk 3]
```diff
@@ -28,9 +28,9 @@ const handler: PlasmoMessaging.MessageHandler = async (req, res) => {
 
   const temporaryPopup = response.data
   temporaryPopup.popup_position = "center"
-  await backgroundPopupCreate([temporaryPopup])
+  // await backgroundPopupCreate([temporaryPopup])
```
Comment (nichgalzin) [inline src/background/messages/viewSinglePopup.ts:31]: Remove coment

**Review by nichgalzin — COMMENTED** [review]
-Sometime they popup twice. Oh no!

**Review by nichgalzin — APPROVED** [review]

**Pull request merged**